### PR TITLE
AOSL-63 Add in code documentation and commit code from staging

### PR DESCRIPTION
**Pull request by seamuslee001**
…and also add in in line code documentation indicating what various bits of code are doing

### CODE REVIEW

File: CRM/Aoservicelisting/Form/ProviderApplicationConfirm.php [hunk 1]
```diff
@@ -335,26 +347,28 @@ public function submit($values)
...
+          $params['master_id'] = $details[0];
           $params['add_relationship'] = 0;
           $params['update_current_employer'] = 0;
           civicrm_api3('Address', 'create', $params);
-        }
 
-        //Ensure that the location type of all email addresses are work.
-        self::setEmailsToWorkLocation($abaMember['id']);
-        // TODO :create website, Do we need to inherit website from the Staff N to ABA Staff N?
+          //Ensure that the location type of all email addresses are work.
```
Comment (seamuslee001) [inline CRM/Aoservicelisting/Form/ProviderApplicationConfirm.php:366]: @Edzelopez you seem to have done this on the staging site but not committed

File: CRM/Aoservicelisting/Form/ProviderApplicationForm.php [hunk 2]
```diff
@@ -40,25 +42,30 @@ public function setDefaultValues() {
...
           $defaults['staff_last_name['. $staffRowCount . ']'] = $defaults['primary_last_name'] = $primaryContact['last_name'];
           $defaults['staff_contact_id['. $staffRowCount . ']'] = $this->_loggedInContactID;
           $defaults['staff_record_regulator[' . $staffRowCount . ']'] = $primaryContact[REGULATED_URL];
           $staffRowCount++;
         }
+        // if the primary contact has a ABA certificate field set then prefill the first ABA staff member details with teh primary contact details. 
         if (!empty($primaryContact[CERTIFICATE_NUMBER])) {
-          $defaults['aba_first_name[' . $abaStaffCount . ']'] = $primaryContact['first_name'];
-          $defaults['aba_last_name[' . $abaStaffCount . ']'] = $primaryContact['last_name'];
+          $defaults['aba_first_name[' . $abaStaffCount . ']'] = $defaults['primary_first_name'] = $primaryContact['first_name'];
```
Comment (seamuslee001) [inline CRM/Aoservicelisting/Form/ProviderApplicationForm.php:55]: @Edzelopez oh look another bug, if the primary contact had been an ABA staff member the primary details wouldn't have filled in on edit i suspect

File: CRM/Aoservicelisting/Form/ProviderApplicationForm.php [hunk 3]
```diff
@@ -322,7 +332,7 @@ public function providerFormRule($values) {
         }
       }
     }
-    $options = self::_getServieOptions();
+    $options = self::_getServiceOptions();
```
Comment (seamuslee001) [inline CRM/Aoservicelisting/Form/ProviderApplicationForm.php:335]: typo in the function name and doing a grep showed this was the only place it was used

File: CRM/Aoservicelisting/Form/ProviderApplicationConfirm.php [hunk 4]
```diff
@@ -338,28 +347,28 @@ public function submit($values)
...
+        // create address for ABA staff as long as they are not primary contact as that will be dealt with lower down. 
         foreach ($addressIds as $key => $details) {
           if ($abaMember['id'] == $primaryContactId) {
             continue;
           }
           $params = $details[1];
           unset($params['id']);
           $params['contact_id'] = $abaMember['id'];
-          $params['master_id'] = $addressIds[$addressKey][0];
+          $params['master_id'] = $details[0];
```
Comment (seamuslee001) [inline CRM/Aoservicelisting/Form/ProviderApplicationConfirm.php:361]: @Edzelopez this wasn't in your other PR but was on staging but i think should be ok

Comment (Edzelopez) [reply CRM/Aoservicelisting/Form/ProviderApplicationConfirm.php:361]: Good catch

File: CRM/Aoservicelisting/Form/ProviderApplicationForm.php (hunk 2 above)
Comment (Edzelopez) [reply CRM/Aoservicelisting/Form/ProviderApplicationForm.php:55]: yup, good catch

File: CRM/Aoservicelisting/Form/ProviderApplicationForm.php (hunk 3 above)
Comment (Edzelopez) [reply CRM/Aoservicelisting/Form/ProviderApplicationForm.php:335]: I've noticed this a long time ago but was too scared to change. Good fix!

**Review by Edzelopez — APPROVED** [review]
Changes look good, and comments appear useful

**Pull request merged**